sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1407,6 +1408,87 @@ object ReplaceExceptWithAntiJoin extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Replaces logical [[Intersect]] operator using a combination of Union, Aggregate
+ * and Generate operator.
+ *
+ * Input Query :
+ * {{{
+ *    SELECT c1 FROM ut1 INTERSECT ALL SELECT c1 FROM ut2
+ * }}}
+ *
+ * Rewritten Query:
+ * {{{
+ *   SELECT c1
+ *   FROM (
+ *        SELECT replicate_row(min_count, c1) AS (min_count, c1)
+ *        FROM (
+ *             SELECT c1,
+ *                    vcol1_cnt,
+ *                    vcol2_cnt,
```

Comment:
Do we need to have `vcol1_cnt` and `vcol2_cnt` here? I think above `replicate_row` only takes `min_count` input.

Reply:
@viirya Thanks !! No we don't. In the actual code, we don't project these columns out. I will fix the doc.